src/main/java/org/openstreetmap/atlas/checks/validation/linear/edges/SinkIslandCheck.java

```diff
@@ -146,9 +154,19 @@ public boolean validCheckForObject(final AtlasObject object)
         // process all those edges
         explored.forEach(marked -> this.markAsFlagged(marked.getIdentifier()));
 
-        // Create the flag if and only if the empty flag value is not set to false
-        return emptyFlag ? Optional.empty()
-                : Optional.of(createFlag(explored, this.getLocalizedInstruction(0)));
+        // We want to filter out any Edges whose end Node has an amenity tag equal to one of the
+        // amenityValuesToExclude
+        final Set<AtlasObject> filteredExploredEdges = explored.stream()
+                .filter(edge -> !endNodeHasAmenityTypeToExclude(edge)).collect(Collectors.toSet());
```

Comment:
I'd suggest using `validEdge` and baking this logic in there. That method is used to filter out possible false positives.